Conflict type lsame_ when include blis.h and lapacke.h in a same file

I don't know if this is expected, but blis.h redifines the `lsame_` auxilary routine of Lapacke : 

$ cat test_blis_lapacke.c 
#include <stdio.h>
#include <lapacke/lapacke.h>
#include <blis.h>

int main()
{
	printf("Hello world\n");
	return 0;
}

$ gcc -std=c99 -I<BLIS_DIR_HERE>/include/blis  test_blis_lapacke.c -o test_blis_lapacke
In file included from /work1/mqho/blis/build/x86/include/blis/blis.h:68:0,
                 from test_blis_lapacke.c:3:
/work1/mqho/blis/build/x86/include/blis/bla_lsame.h:37:23: error: conflicting types for ‘lsame_’
 bla_logical PASTEF770(lsame)(const bla_character *ca, const bla_character *cb, ftnlen ca_len, ftnlen cb_len);
                       ^
/work1/mqho/blis/build/x86/include/blis/bli_macro_defs.h:111:52: note: in definition of macro ‘PASTEF770’
 #define PASTEF770(name)                            name ## _
                                                    ^
In file included from /usr/include/lapacke/lapacke.h:143:0,
                 from test_blis_lapacke.c:2:
/usr/include/lapacke/lapacke.h:146:16: note: previous declaration of ‘lsame_’ was here
 lapack_logical LAPACK_lsame( char* ca,  char* cb,
                ^

--- a/frame/compat/f2c/bla_lsame.c
+++ b/frame/compat/f2c/bla_lsame.c
@@ -41,7 +41,12 @@
 	-lf2c -lm   (in that order)
 */
 
-bla_logical PASTEF770(lsame)(const bla_character *ca, const bla_character *cb, ftnlen ca_len, ftnlen cb_len)
+
+#ifdef LAPACK_ILP64
+long PASTEF770(lsame)(char *ca, char *cb, long ca_len, long cb_len)
+#else
+int PASTEF770(lsame)(char *ca, char *cb, int ca_len, int cb_len)
+#endif
 {
     /* System generated locals */
     bla_logical ret_val;
@@ -115,11 +120,11 @@ bla_logical PASTEF770(lsame)(const bla_character *ca, const bla_character *cb, f
 /*        EBCDIC is assumed - ZCODE is the EBCDIC code of either lower or */
 /*        upper case 'Z'. */
 
-	if ((inta >= 129 && inta <= 137) || (inta >= 145 && inta <= 153) || (inta 
+	if ((inta >= 129 && inta <= 137) || (inta >= 145 && inta <= 153) || (inta
 		>= 162 && inta <= 169)) {
 	    inta += 64;
 	}
-	if ((intb >= 129 && intb <= 137) || (intb >= 145 && intb <= 153) || (intb 
+	if ((intb >= 129 && intb <= 137) || (intb >= 145 && intb <= 153) || (intb
 		>= 162 && intb <= 169)) {
 	    intb += 64;
 	}

--- a/frame/compat/f2c/bla_lsame.h
+++ b/frame/compat/f2c/bla_lsame.h
@@ -34,6 +34,10 @@
 
 #ifdef BLIS_ENABLE_BLAS2BLIS
 
-bla_logical PASTEF770(lsame)(const bla_character *ca, const bla_character *cb, ftnlen ca_len, ftnlen cb_len);
+#ifdef LAPACK_ILP64
+long PASTEF770(lsame)(char *ca, char *cb, long ca_len, long cb_len);
+#else
+int PASTEF770(lsame)(char *ca, char *cb, int ca_len, int cb_len);
+#endif
 
 #endif

